typescript/src/redemption.ts

```diff
@@ -0,0 +1,291 @@
+// @ts-ignore
+import bcoin from "bcoin"
+// @ts-ignore
+import hash160 from "bcrypto/lib/hash160"
+import { BigNumber } from "ethers"
+import {
+  createKeyRing,
+  RawTransaction,
+  UnspentTransactionOutput,
+  Client as BitcoinClient,
+} from "./bitcoin"
+import { Bridge } from "./bridge"
+
+/**
+ * Represents a redemption request.
+ */
+export interface RedemptionRequest {
+  /**
+   * The address of the redeemer.
+   */
+  redeemerAddress: string
+
+  /**
+   * The amount of Bitcoins in satoshis that is requested to be redeemed.
+   * The actual value of the output in the Bitcoin transaction will be decreased
+   * by the sum of the fee share and the treasury fee for this particular output.
+   */
+  requestedAmount: BigNumber
+
+  /**
+   * The amount of Bitcoins in satoshis that is subtracted from the amount of
+   * the redemption request and used to pay the treasury fee.
+   * The value should be exactly equal to the value of treasury fee in the Bridge
+   * on-chain contract at the time the redemption request was made.
+   */
+  treasuryFee: BigNumber
+
+  /**
+   * The amount of Bitcoins in satoshis that is subtracted from the amount of
+   * the redemption request and used to pay the transaction network fee.
+   * The value must not exceed the value of the Bridge's txMaxFee parameter
+   * that was in force at the moment the redemption request was made. The sum
+   * of the txFee values taken from all redemption outputs is equivalent to
+   * the total network fee of the entire redemption transaction.
+   */
+  txFee: BigNumber
```

Comment:
Step 2:
I think it makes more sense to have just the `txMaxFee` parameter here. The actual fee used should be determined during `createRedemptionTransaction` and embedded in the Bitcoin transaction.

Reply:
Done in https://github.com/keep-network/tbtc-v2/pull/199/commits/4b3be0a5f176ab3e5733f1876ff60fa6d3ca53a2.